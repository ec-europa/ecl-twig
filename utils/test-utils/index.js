const Twig = require('twig');
const { format } = require('prettier');

const renderTwigFile = (file, options, cb) =>
  Twig.renderFile(file, options, (err, html) =>
    cb(err, format(html, { parser: 'html' }))
  );

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) =>
    Twig.renderFile(file, options, (err, html) => {
      if (err) return reject(err);

      const div = document.createElement('div');
      div.innerHTML = html.trim();

      return resolve(div.firstChild);
    })
  );

module.exports = {
  renderTwigFile,
  renderTwigFileAsNode,
};
