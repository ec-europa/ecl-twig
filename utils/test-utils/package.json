{
  "name": "@ecl-twig/test-utils",
  "version": "2.29.0",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
