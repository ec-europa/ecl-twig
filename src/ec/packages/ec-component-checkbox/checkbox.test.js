import { merge, renderTwigFileAsNode } from '@ecl-twig/test-utils';

// Import data for tests
import { dataDefault, dataInvalid } from './demo/data';

describe('EC - checkbox', () => {
  const template = '@ecl-twig/ec-component-checkbox/checkbox-group.html.twig';
  const render = params => renderTwigFileAsNode(template, params);

  describe('Default', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(render(dataDefault)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(dataDefault, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(dataDefault, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });
  });

  describe('Invalid', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(render(dataInvalid)).resolves.toMatchSnapshot();
    });
  });
});
