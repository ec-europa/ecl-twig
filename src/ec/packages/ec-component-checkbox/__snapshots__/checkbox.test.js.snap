// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EC - checkbox Default renders correctly 1`] = `
<fieldset
  aria-describedby="checkbox-default-helper"
  class="ecl-form-group ecl-form-group--checkbox"
>
  <div
    class="ecl-checkbox__help ecl-help-block"
    id="checkbox-default-helper"
  >
    Helper text for the group
  </div>
  <div
    class="ecl-checkbox"
  >
    <input
      class="ecl-checkbox__input"
      id="checkbox-default-1"
      name="checkbox-default"
      type="checkbox"
      value="es"
    />
    <label
      class="ecl-form-label ecl-checkbox__label"
      for="checkbox-default-1"
    >
      <span
        class="ecl-checkbox__box"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--s ecl-checkbox__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#ui--check"
          />
        </svg>
      </span>
      Spain
    </label>
    <p
      class="ecl-checkbox__help ecl-help-block"
      id="helper-default-1"
    >
      Help text for option 1
    </p>
  </div>
  <div
    class="ecl-checkbox"
  >
    <input
      class="ecl-checkbox__input"
      id="checkbox-default-2"
      name="checkbox-default"
      type="checkbox"
      value="be"
    />
    <label
      class="ecl-form-label ecl-checkbox__label"
      for="checkbox-default-2"
    >
      <span
        class="ecl-checkbox__box"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--s ecl-checkbox__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#ui--check"
          />
        </svg>
      </span>
      Belgium
    </label>
    <p
      class="ecl-checkbox__help ecl-help-block"
      id="helper-default-2"
    >
      Help text for option 2
    </p>
  </div>
  <div
    class="ecl-checkbox ecl-checkbox--disabled"
  >
    <input
      class="ecl-checkbox__input"
      disabled=""
      id="checkbox-default-3"
      name="checkbox-default"
      type="checkbox"
      value="fr"
    />
    <label
      class="ecl-form-label ecl-checkbox__label"
      for="checkbox-default-3"
    >
      <span
        class="ecl-checkbox__box"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--s ecl-checkbox__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#ui--check"
          />
        </svg>
      </span>
      France (disabled)
    </label>
    <p
      class="ecl-checkbox__help ecl-help-block ecl-help-block--disabled"
      id="helper-default-3"
    >
      Help text for option 3
    </p>
  </div>
</fieldset>
`;

exports[`EC - checkbox Default renders correctly when required 1`] = `
<fieldset
  aria-describedby="checkbox-default-helper"
  class="ecl-form-group ecl-form-group--checkbox"
>
  <div
    class="ecl-checkbox__help ecl-help-block"
    id="checkbox-default-helper"
  >
    Helper text for the group
  </div>
  <div
    class="ecl-checkbox"
  >
    <input
      class="ecl-checkbox__input"
      id="checkbox-default-1"
      name="checkbox-default"
      required=""
      type="checkbox"
      value="es"
    />
    <label
      class="ecl-form-label ecl-checkbox__label"
      for="checkbox-default-1"
    >
      <span
        class="ecl-checkbox__box"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--s ecl-checkbox__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#ui--check"
          />
        </svg>
      </span>
      Spain
    </label>
    <p
      class="ecl-checkbox__help ecl-help-block"
      id="helper-default-1"
    >
      Help text for option 1
    </p>
  </div>
  <div
    class="ecl-checkbox"
  >
    <input
      class="ecl-checkbox__input"
      id="checkbox-default-2"
      name="checkbox-default"
      required=""
      type="checkbox"
      value="be"
    />
    <label
      class="ecl-form-label ecl-checkbox__label"
      for="checkbox-default-2"
    >
      <span
        class="ecl-checkbox__box"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--s ecl-checkbox__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#ui--check"
          />
        </svg>
      </span>
      Belgium
    </label>
    <p
      class="ecl-checkbox__help ecl-help-block"
      id="helper-default-2"
    >
      Help text for option 2
    </p>
  </div>
  <div
    class="ecl-checkbox ecl-checkbox--disabled"
  >
    <input
      class="ecl-checkbox__input"
      disabled=""
      id="checkbox-default-3"
      name="checkbox-default"
      required=""
      type="checkbox"
      value="fr"
    />
    <label
      class="ecl-form-label ecl-checkbox__label"
      for="checkbox-default-3"
    >
      <span
        class="ecl-checkbox__box"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--s ecl-checkbox__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#ui--check"
          />
        </svg>
      </span>
      France (disabled)
    </label>
    <p
      class="ecl-checkbox__help ecl-help-block ecl-help-block--disabled"
      id="helper-default-3"
    >
      Help text for option 3
    </p>
  </div>
</fieldset>
`;

exports[`EC - checkbox Default renders correctly with extra attributes 1`] = `
<fieldset
  aria-describedby="checkbox-default-helper"
  class="ecl-form-group ecl-form-group--checkbox"
  data-test="data-test-value"
  data-test-1="data-test-value-1"
>
  <div
    class="ecl-checkbox__help ecl-help-block"
    id="checkbox-default-helper"
  >
    Helper text for the group
  </div>
  <div
    class="ecl-checkbox"
  >
    <input
      class="ecl-checkbox__input"
      id="checkbox-default-1"
      name="checkbox-default"
      type="checkbox"
      value="es"
    />
    <label
      class="ecl-form-label ecl-checkbox__label"
      for="checkbox-default-1"
    >
      <span
        class="ecl-checkbox__box"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--s ecl-checkbox__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#ui--check"
          />
        </svg>
      </span>
      Spain
    </label>
    <p
      class="ecl-checkbox__help ecl-help-block"
      id="helper-default-1"
    >
      Help text for option 1
    </p>
  </div>
  <div
    class="ecl-checkbox"
  >
    <input
      class="ecl-checkbox__input"
      id="checkbox-default-2"
      name="checkbox-default"
      type="checkbox"
      value="be"
    />
    <label
      class="ecl-form-label ecl-checkbox__label"
      for="checkbox-default-2"
    >
      <span
        class="ecl-checkbox__box"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--s ecl-checkbox__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#ui--check"
          />
        </svg>
      </span>
      Belgium
    </label>
    <p
      class="ecl-checkbox__help ecl-help-block"
      id="helper-default-2"
    >
      Help text for option 2
    </p>
  </div>
  <div
    class="ecl-checkbox ecl-checkbox--disabled"
  >
    <input
      class="ecl-checkbox__input"
      disabled=""
      id="checkbox-default-3"
      name="checkbox-default"
      type="checkbox"
      value="fr"
    />
    <label
      class="ecl-form-label ecl-checkbox__label"
      for="checkbox-default-3"
    >
      <span
        class="ecl-checkbox__box"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--s ecl-checkbox__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#ui--check"
          />
        </svg>
      </span>
      France (disabled)
    </label>
    <p
      class="ecl-checkbox__help ecl-help-block ecl-help-block--disabled"
      id="helper-default-3"
    >
      Help text for option 3
    </p>
  </div>
</fieldset>
`;

exports[`EC - checkbox Default renders correctly with extra class names 1`] = `
<fieldset
  aria-describedby="checkbox-default-helper"
  class="ecl-form-group ecl-form-group--checkbox custom-class custom-class--test"
>
  <div
    class="ecl-checkbox__help ecl-help-block"
    id="checkbox-default-helper"
  >
    Helper text for the group
  </div>
  <div
    class="ecl-checkbox"
  >
    <input
      class="ecl-checkbox__input"
      id="checkbox-default-1"
      name="checkbox-default"
      type="checkbox"
      value="es"
    />
    <label
      class="ecl-form-label ecl-checkbox__label"
      for="checkbox-default-1"
    >
      <span
        class="ecl-checkbox__box"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--s ecl-checkbox__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#ui--check"
          />
        </svg>
      </span>
      Spain
    </label>
    <p
      class="ecl-checkbox__help ecl-help-block"
      id="helper-default-1"
    >
      Help text for option 1
    </p>
  </div>
  <div
    class="ecl-checkbox"
  >
    <input
      class="ecl-checkbox__input"
      id="checkbox-default-2"
      name="checkbox-default"
      type="checkbox"
      value="be"
    />
    <label
      class="ecl-form-label ecl-checkbox__label"
      for="checkbox-default-2"
    >
      <span
        class="ecl-checkbox__box"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--s ecl-checkbox__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#ui--check"
          />
        </svg>
      </span>
      Belgium
    </label>
    <p
      class="ecl-checkbox__help ecl-help-block"
      id="helper-default-2"
    >
      Help text for option 2
    </p>
  </div>
  <div
    class="ecl-checkbox ecl-checkbox--disabled"
  >
    <input
      class="ecl-checkbox__input"
      disabled=""
      id="checkbox-default-3"
      name="checkbox-default"
      type="checkbox"
      value="fr"
    />
    <label
      class="ecl-form-label ecl-checkbox__label"
      for="checkbox-default-3"
    >
      <span
        class="ecl-checkbox__box"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--s ecl-checkbox__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#ui--check"
          />
        </svg>
      </span>
      France (disabled)
    </label>
    <p
      class="ecl-checkbox__help ecl-help-block ecl-help-block--disabled"
      id="helper-default-3"
    >
      Help text for option 3
    </p>
  </div>
</fieldset>
`;

exports[`EC - checkbox Invalid renders correctly 1`] = `
<fieldset
  aria-describedby="checkbox-default-helper"
  class="ecl-form-group ecl-form-group--checkbox"
>
  <div
    class="ecl-checkbox__help ecl-help-block"
    id="checkbox-default-helper"
  >
    Helper text for the group
  </div>
  <div
    class="ecl-feedback-message"
  >
    Error message for the group
  </div>
  <div
    class="ecl-checkbox ecl-checkbox--invalid"
  >
    <input
      class="ecl-checkbox__input"
      id="checkbox-invalid-1"
      name="checkbox-invalid"
      type="checkbox"
      value="es"
    />
    <label
      class="ecl-form-label ecl-checkbox__label ecl-form-label--invalid"
      for="checkbox-invalid-1"
    >
      <span
        class="ecl-checkbox__box ecl-checkbox__box--invalid"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--s ecl-checkbox__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#ui--check"
          />
        </svg>
      </span>
      Spain
    </label>
    <p
      class="ecl-checkbox__help ecl-help-block"
      id="helper-invalid-1"
    >
      Help text for option 1
    </p>
  </div>
  <div
    class="ecl-checkbox ecl-checkbox--invalid"
  >
    <input
      class="ecl-checkbox__input"
      id="checkbox-invalid-2"
      name="checkbox-invalid"
      type="checkbox"
      value="be"
    />
    <label
      class="ecl-form-label ecl-checkbox__label ecl-form-label--invalid"
      for="checkbox-invalid-2"
    >
      <span
        class="ecl-checkbox__box ecl-checkbox__box--invalid"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--s ecl-checkbox__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#ui--check"
          />
        </svg>
      </span>
      Belgium
    </label>
    <p
      class="ecl-checkbox__help ecl-help-block"
      id="helper-invalid-2"
    >
      Help text for option 2
    </p>
  </div>
  <div
    class="ecl-checkbox ecl-checkbox--invalid ecl-checkbox--disabled"
  >
    <input
      class="ecl-checkbox__input"
      disabled=""
      id="checkbox-invalid-3"
      name="checkbox-invalid"
      type="checkbox"
      value="fr"
    />
    <label
      class="ecl-form-label ecl-checkbox__label ecl-form-label--invalid"
      for="checkbox-invalid-3"
    >
      <span
        class="ecl-checkbox__box ecl-checkbox__box--invalid"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--s ecl-checkbox__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#ui--check"
          />
        </svg>
      </span>
      France (disabled)
    </label>
    <p
      class="ecl-checkbox__help ecl-help-block ecl-help-block--disabled"
      id="helper-invalid-3"
    >
      Help text for option 3
    </p>
  </div>
</fieldset>
`;
